packages/salesforcedx-vscode-core/test/commands/forceProjectCreate.test.ts

```diff
@@ -0,0 +1,165 @@
+import { expect } from 'chai';
+import * as path from 'path';
+import * as sinon from 'sinon';
+import * as vscode from 'vscode';
+import { ContinueResponse } from '../../src/commands/commands';
+import {
+  ForceProjectCreateExecutor,
+  PathExistsChecker,
+  ProjectNameAndPath,
+  SelectProjectFolder,
+  SelectProjectName
+} from '../../src/commands/forceProjectCreate';
+import { nls } from '../../src/messages';
+
+// tslint:disable:no-unused-expression
+describe('Force Project Create', () => {
+  const PROJECT_NAME = 'sfdx-simple';
+  const WORKSPACE_PATH = path.join(vscode.workspace.rootPath!, '..');
+  const PROJECT_DIR: vscode.Uri[] = [vscode.Uri.parse(WORKSPACE_PATH)];
+
+  describe('SelectProjectName Gatherer', () => {
+    let inputBoxSpy: sinon.SinonStub;
+
+    before(() => {
+      inputBoxSpy = sinon.stub(vscode.window, 'showInputBox');
+      inputBoxSpy.onCall(0).returns(undefined);
+      inputBoxSpy.onCall(1).returns('');
+      inputBoxSpy.onCall(2).returns(PROJECT_NAME);
+    });
+
+    after(() => {
+      inputBoxSpy.restore();
+    });
+
+    it('Should return cancel if project name is undefined', async () => {
+      const gatherer = new SelectProjectName();
+      const response = await gatherer.gather();
+      expect(inputBoxSpy.calledOnce).to.be.true;
+      expect(response.type).to.equal('CANCEL');
+    });
+
+    it('Should return cancel if user input is empty string', async () => {
```

Comment:
Try a string with just white spaces also.

Reply:
@kaplanjosh @DatGuyJonathan Should we allow a user to enter just white space? If not, do we show them some form of info/error message? The CLI command works with just white space as a project name. I think we should prompt if they want to continue if the name is all whitespace but open to other ideas if y'all have any.

Reply:
![image](https://user-images.githubusercontent.com/4503716/32512087-af766eaa-c3aa-11e7-94e5-722621f7f13f.png)

Reply:
I see, so force-language-services allows it. It could be by-design that we don't trim the name; @GaryFNC can offer his thoughts on that. If anything is going to be fixed, it should go into force-language-services .

Reply:
The original discussion with Mike Miller, etc.  was that any valid "folder" name was OK - so not much checking in force-language-services....   BUT, it shouldn't really allow empty input or just white spaces.....     That should be caught in FLS, but it is a bit harder for a user to input that on the CLI command line, so didn't get noticed.    We don't generally trim the names since spaces are allowed in folder names (although yet, it is odd and hard to read if the spaces all come at the end).